Client/Frontend/Shields/ShareShieldsController.swift

```diff
@@ -0,0 +1,280 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import BraveShared
+import Shared
+import BraveUI
+
+// MARK: - ShareShieldsViewController
+
+class ShareShieldsViewController: UIViewController, Themeable {
+    
+    // MARK: Action
+    
+    enum Action {
+        case shareEmailClicked
+        case shareTwitterClicked
+        case shareFacebookClicked
+        case shareDefaultClicked
+    }
+    
+    // MARK: UX
+    
+    private struct UX {
+        static let defaultInset: UIEdgeInsets = UIEdgeInsets(top: 19, left: 19, bottom: 19, right: 19)
+        static let contentSizeChange: CGFloat = 64
+    }
+    
+    // MARK: Properties
+    
+    private let tab: Tab
+
+    private let contentStackView = UIStackView().then {
+        $0.axis = .vertical
+        $0.spacing = 3
+        $0.alignment = .fill
+        $0.layoutMargins = UX.defaultInset
+        $0.isLayoutMarginsRelativeArrangement = true
+    }
+    
+    var actionHandler: ((Action) -> Void)?
+    
+    // MARK: Lifecycle
+    
+    init(tab: Tab) {
+        self.tab = tab
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    @available(*, unavailable)
+    required init(coder: NSCoder) {
+        fatalError()
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = Strings.Share.shareScreenTitle
+    
+        doLayout()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        preferredContentSize = contentStackView.systemLayoutSizeFitting(
+            CGSize(width: view.bounds.size.width - UX.contentSizeChange,
+                   height: view.bounds.size.height - UX.contentSizeChange),
+            withHorizontalFittingPriority: .required,
+            verticalFittingPriority: .fittingSizeLevel
+        ).with {
+            $0.width += UX.contentSizeChange
+        }
+        
+        navigationController?.preferredContentSize = preferredContentSize
+    }
+        
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        
+        if #available(iOS 13.0, *) {
+            if traitCollection.userInterfaceStyle != previousTraitCollection?.userInterfaceStyle {
+                applyTheme(Theme.of(tab))
+            }
+        }
+    }
+    
+    // MARK: Internal
+    
+    private func doLayout() {
```

Comment:
This function adds gesture recognizers too, maybe rename to `doLayoutAndAddGestureRecognizers` or something better